runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningPlugin.java

```diff
@@ -161,6 +203,68 @@ public void onWallObjectSpawned(WallObjectSpawned event)
 		}
 	}
 
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		if (event.getWidgetId() != WidgetInfo.INVENTORY.getId() || event.getMenuAction() == MenuAction.RUNELITE)
+		{
+			return;
+		}
+
+		ItemContainer inventoryItemContainer = client.getItemContainer(InventoryID.INVENTORY);
+
+		switch (event.getMenuOption().toLowerCase())
+		{
+			case FILL_OPTION:
+				updateAmountOfCoalInBag((int) Arrays.stream(inventoryItemContainer.getItems()).filter(i -> i.getId() == ItemID.COAL).count());
+				break;
+			case EMPTY_OPTION:
+				int amt = MAX_INVENTORY_SPACE - (int) Arrays.stream(inventoryItemContainer.getItems()).filter(i -> i.getId() != -1).count();
+				updateAmountOfCoalInBag(-amt);
+				break;
+		}
+	}
+
+	/**
+	 * Use onChatMessage when a player chooses the `Check` menu option on coal bags.
+	 *
+	 * @param event
+	 */
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		String chatMsg = Text.removeTags(event.getMessage());
+		if (COAL_BAG_EMPTY_MESSAGE.matcher(chatMsg).find())
```

Comment:
This needs to check the chat message type or else people can type things in public chat and trigger the plugin.